policyengine_us/variables/gov/states/hi/tax/income/computation/hi_alternative_tax_on_capital_gains.py

```diff
@@ -0,0 +1,63 @@
+from policyengine_us.model_api import *
+
+
+class hi_alternative_tax_on_capital_gains(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Hawaii alternative tax on capital gains"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.HI
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        p = parameters(period).gov.states.hi.tax.income.alternative_tax
+        filing_status = tax_unit("filing_status", period)
+        taxable_income = tax_unit("hi_taxable_income", period)
+        net_lt_capital_gain = add(person, period, ["long_term_capital_gains"])
+        net_capital_gain = tax_unit("net_capital_gain", period)
+        # line_9 how to calculate Net gain from the disposition of property held for investment
+        cap = p.max_amount[filing_status]
+        # income_tax
+        income_tax = tax_unit("hi_income_tax_before_credits", period)
+        eligibility = tax_unit(
+            "hi_alternative_tax_on_capital_gains_eligible", period
+        )
+
+        # line 13
+        ineligible_income = max_(
+            taxable_income
+            - min_(net_capital_gain, net_lt_capital_gain),  # line 11
+            cap,
+        )
+
+        # What if line 1 < line 13
+        eligible_income = max_(
+            0, taxable_income - ineligible_income
+        )  # net capital gains eligble for alternative tax
+
+        # ineligible_tax --- tax for line 13
+        statuses = filing_status.possible_values
+        rate_p = parameters(period).gov.states.hi.tax.income.rates
+        ineligible_tax = select(
+            [
+                filing_status == statuses.SINGLE,
+                filing_status == statuses.SEPARATE,
+                filing_status == statuses.JOINT,
+                filing_status == statuses.WIDOW,
+                filing_status == statuses.HEAD_OF_HOUSEHOLD,
+            ],
+            [
+                rate_p.single.calc(ineligible_income),
+                rate_p.separate.calc(ineligible_income),
+                rate_p.joint.calc(ineligible_income),
+                rate_p.widow.calc(ineligible_income),
+                rate_p.head_of_household.calc(ineligible_income),
+            ],
+        )
+
+        alternative_tax = ineligible_tax + eligible_income * p.rate  # line 17
+
+        return where(
```

Comment:
I think we should have this variable be just the calculation for the alternative minimum tax

Reply:
dont need the income tax coparison

Reply:
![image](https://github.com/PolicyEngine/policyengine-us/assets/110687043/97e5cbdc-e61d-486d-bd93-2e20e7d5bb0b)

we want to include it in the hawaii taxable income variable

Reply:
Got it. I found that in the worksheet for tax on capital gains, the final step is like this:
![image](https://github.com/PolicyEngine/policyengine-us/assets/97819588/d7df3739-429e-4b75-8732-86febe6ff341)
Here line 17 is the adjusted tax with alternative tax on capital gains, and line 18 is the original income tax for the taxable income. It says if line 17 is smaller, we should enter line 14, which is the amount of net capital gains eligible for alternative tax here. I want to ask, if line 17 is larger than line 18, which means that after substituting the tax for eligible parts with alternative tax on capital gains, the final tax becomes larger, then what value should we return, should we still return the amount of net captital gains eligible for alternative tax? Thank you!

Reply:
We want to return line 17 for this variable hi_alternative_tax_on_capital_gains

and then in the variable hi_income_tax_before_credits we want to make the comparison to from original income tax (line 18) with the alternative (line 17) 

The entering of line 14 into the form does not have any computational purposes (it is not used anywhere else) so we can just ignore that

Reply:
Got it! It's rather clear. Thank you！

Reply:
Thank you for the comment, it's rather clear. By the way, I have a further question about the return value for the hi_alternative_tax_on_capital_gains. If the filer is not eligible for the alternative tax, since we check the filer the eligibility first, what value should we return then or we just care about the eligible situation in this variable?

Reply:
The problem is that if we return 0 for ineligible filers then we always get 0 for the tax computation as that is the smallest.

For ineligible filers we want to return the original tax value as they will be selecting that over the hi_alternative_tax_on_capital_gains


let me know if that makes sense or if I missed the point